### PR TITLE
Build: Clean-up the project target frameworks, remove unnecessary build legs, replace bootstrap with project specific PackageDownload

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/841

Regression? Last working version:

## Description

* Clean-up the target frameworks in a couple of the test projects, as they had extra frameworks (ie, tests had 3 frameworks, but product code has 2). See old and new framework list: https://raw.githubusercontent.com/NuGet/NuGet.Client/f3904be83daaf8a333fc210d51e17acd288a0761/old.md and https://raw.githubusercontent.com/NuGet/NuGet.Client/f3904be83daaf8a333fc210d51e17acd288a0761/new.md. 
* Remove unnecessary TestProject declaration. They are already covered.
* Further cleanup the target framework declarations, by making them *easier* to change. 
* Replace some of the bootstrap packages with PackageDownload instead, further moving us away from needing configure.ps1 for building locally. 

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: build/common.targets [hunk 1]
```diff
@@ -391,6 +391,15 @@ Condition=" ('%(Extension)' == '.dll' OR '%(Filename)' == 'NuGet.CommandLine.XPl
       Condition=" '%(PackageReference.Version)' != '' AND '%(PackageReference.IsImplicitlyDefined)' != 'true' AND '%(PackageReference.Identity)' != 'Microsoft.NET.Test.Sdk' " />
   </Target>
 
+  <Target Name="GetAllTargetFrameworks">
```
Comment (zivkan) [inline build/common.targets]: I didn't see this being used anywhere. Is it needed? What's it used for?

File: src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj [hunk 2]
```diff
@@ -83,11 +83,12 @@
     <Reference Include="System.Xml" />
     <Reference Include="System.Xml.Linq" />
     <Reference Include="NuGet.Core">
-      <HintPath>$(SolutionPackagesFolder)nuget.core\2.14.0-rtm-832\lib\net40-Client\NuGet.Core.dll</HintPath>
+      <HintPath>$(NuGetPackageRoot)nuget.core\2.14.0-rtm-832\lib\net40-Client\NuGet.Core.dll</HintPath>
```
Comment (zivkan) [inline src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj]: One of the advantages of `packages.targets` is the the versions are usually specified in a single place. Could we have a `NuGetCoreVersion` property set there, and use `<HintPath>$(NuGetPackageRoot)nuget.core\$(NuGetCoreVersion)\...`? Similar for the other assembly references from other `PackageDownload` in other files?

**Review by zivkan — APPROVED** [review]

File: build/common.targets (hunk 1 above)
Comment (nkolev92) [reply build/common.targets]: I meant to remove that. 
I used it to generate the list of frameworks (unfortunately the project overview doesn't have the frameworks).

File: src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj (hunk 2 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj]: Yep, good idea. I'll do that.

**Pull request merged**